src/base/preferences.cpp

```diff
@@ -304,6 +304,45 @@ void Preferences::setIconsInMenusEnabled(const bool enable)
 }
 #endif // Q_OS_MACOS
 
+qint64 Preferences::getTorrentFileSizeLimit() const
+{
+    return value(u"BitTorrent/TorrentFileSizeLimit"_s, (100 * 1024 * 1024));
+}
+
+void Preferences::setTorrentFileSizeLimit(const qint64 value)
+{
+    if (value == getTorrentFileSizeLimit())
+        return;
+
+    setValue(u"BitTorrent/TorrentFileSizeLimit"_s, value);
+}
+
+int Preferences::getBdecodeDepthLimit() const
+{
+    return value(u"BitTorrent/BdecodeDepthLimit"_s, 100);
```

Comment:
Can you suggest a full name?
Having `BencodeDepthLimit` is wrong because it is about decoding. And having `BencodeDecodeDepthLimit` looks confusing because you put encode and decode together. So currently I pick `Bdecode` even it is a shortcut.

Reply:
> Having `BencodeDepthLimit` is wrong because it is about decoding.

Why is it wrong? I don't feel so. We just talk about some aspects of Bencode-formatted data we limit during decoding (depth, number of tokens etc.). We can talk about other formats in the similar way, e.g. `JSONDepthLimit`, `XMLDepthLimit`.

Reply:
However, I do not insist.

Reply:
What about `BencodeDecodeDepthLimit` which I can compromise on? I feel like encoding and decoding limits should be distinct even if libtorrent doesn't support it.
I also I suppose this only matters with the config key name and not the function name.

Reply:
Well, leave it as is. Now I see that "bdecode" seems to be widely used in the wild.
Also it seems that it is easier to match with error messages, libtorrent documentation etc.